core/app/models/spree/stock/allocator/base.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Spree
+  module Stock
+    module Allocator
+      class Base
+        attr_reader :desired, :availability
+
+        def initialize(desired, availability)
+          @desired = desired
+          @availability = availability
+        end
+
+        def allocate_on_hand
```

Comment:
I'm not 100% sure this can be accomplished, but since I don't really like how the previous implementation stored the state of execution in `@desired`, I think it'd be great if we could allocate all inventory in a single method call, e.g.

```ruby
on_hand_packages, backordered_packages = @allocator.allocate(desired)
```

This way the allocator doesn't hold any state apart from the availability, and can be potentially reused multiple times. This won't really happen most likely, but I think it's still a better design than what we have now.

Reply:
@aldesantis Good advice, check if now it's similar to your design 👍